test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -3245,16 +3245,8 @@ TEST(LoopNest, FlattenImperfectLoopNest) {
 
   std::vector<For*> loops = {outer_for, inner_for};
   For* flattened = nullptr;
-  bool success = LoopNest::flatten(loops, &flattened);
-  ASSERT_FALSE(success);
-
-  auto result = IRSimplifier::simplify(flattened);
-  checkIR(result, R"IR(
```

Comment:
I think it might make sense to check IR output at least in one "incorrect" test. Alternatively, we could compute a hash before and after and assert equality.

Reply:
Good idea. Added an equivalence check using hash for all the negative tests. Thanks for pointing that out.